### PR TITLE
Adds data source properties to onedrive, reddit and slack

### PR DESCRIPTION
**Pull request by rvztz**
Addresses #1316.

Adds data source properties to onedrive, reddit and slack:
-  data_created
-  date_modified
-  version
-  record_locator
These properties are instantiated when supported by the connector.

Separates the logic between fetching the file from source and `get_file`. Retrieves file metadata when any of the properties are called.

Adds logic to check if file exists in the remote source. For connectors that don't directly support it, adds exception handling to check any issues while retrieving the file.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@
 * **Better debug output related to sentence counting internals**. Clarify message when sentence is not counted toward sentence count because there aren't enough words, relevant for developers focused on `unstructured`s NLP internals.
 * **Faster ocr_only speed for partitioning PDF and images.** Use `unstructured_pytesseract.run_and_get_multiple_output` function to reduce the number of calls to `tesseract` by half when partitioning pdf or image with `tesseract`
 * **Adds data source properties to fsspec connectors** These properties (date_created, date_modified, version, source_url, record_locator) are written to element metadata during ingest, mapping elements to information about the document source from which they derive.
+* **Adds data source properties (date_created, date_modified, version, exists, source_url, record_locator) to Onedrive, Slack and Reddit connectors.** Implements `update_source_metadata` method to onedrive, reddit and slack connectors.
```
Comment (ryannikolaidis) [inline CHANGELOG.md:15]:
```suggestion
* **Adds data source properties to connectors** These properties (date_created, date_modified, version, source_url, record_locator) are written to element metadata during ingest, mapping elements to information about the document source from which they derive.
```

**Review by ryannikolaidis — APPROVED** [review]
approved!

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,7 +12,7 @@
...
 * **Faster ocr_only speed for partitioning PDF and images.** Use `unstructured_pytesseract.run_and_get_multiple_output` function to reduce the number of calls to `tesseract` by half when partitioning pdf or image with `tesseract`
 * **Adds data source properties to fsspec connectors** These properties (date_created, date_modified, version, source_url, record_locator) are written to element metadata during ingest, mapping elements to information about the document source from which they derive.
-* **Adds data source properties (date_created, date_modified, version, exists, source_url, record_locator) to Onedrive, Slack and Reddit connectors.** Implements `update_source_metadata` method to onedrive, reddit and slack connectors. 
+* **Adds data source properties to connectors** These properties (date_created, date_modified, version, source_url, record_locator) are written to element metadata during ingest, mapping elements to information about the document source from which they derive.
```
Comment (cragwolfe) [inline CHANGELOG.md:15]: why does it matter?

Comment (cragwolfe) [reply CHANGELOG.md:15]: sorry, i mean bullet points should cover "what was broken? what was fixed? why does it matter?" for enhancements, only the last part should be answered

File: CHANGELOG.md [hunk 3]
```diff
@@ -12,6 +12,7 @@
 * **Better debug output related to sentence counting internals**. Clarify message when sentence is not counted toward sentence count because there aren't enough words, relevant for developers focused on `unstructured`s NLP internals.
 * **Faster ocr_only speed for partitioning PDF and images.** Use `unstructured_pytesseract.run_and_get_multiple_output` function to reduce the number of calls to `tesseract` by half when partitioning pdf or image with `tesseract`
 * **Adds data source properties to fsspec connectors** These properties (date_created, date_modified, version, source_url, record_locator) are written to element metadata during ingest, mapping elements to information about the document source from which they derive.
+* **Adds data source properties to connectors** These properties (date_created, date_modified, version, source_url, record_locator) are written to element metadata during ingest, mapping elements to information about the document source from which they derive.
```
Comment (ryannikolaidis) [inline CHANGELOG.md:15]:
```suggestion
**Adds data source properties to Onedrive, Reddit, and Slack connectors** These properties (date_created, date_modified, version, source_url, record_locator) are written to element metadata during ingest, mapping elements to information about the document source from which they derive. This functionality enables downstream applications to reveal source document applications, e.g. a link to a GDrive doc, Salesforce record, etc.
```

**Pull request merged**